### PR TITLE
Migrate Scenarios to TS

### PR DESCRIPTION
**Pull request by BenjiLee**
@lxu213 

This PR migrates Scenarios.js -> Secanrios.tsx

TODOs:
- [x] Self review PR before requesting review

### CODE REVIEW

File: src/components/Filters/Scenarios.tsx [hunk 1]
```diff
@@ -2,9 +2,31 @@ import React, { Component } from 'react';
...
+    SCENARIOS: ScenarioList,
+    scenario: Scenario,
+    onScenarioClick(scenarioList: ScenarioList): () => void,
+    onScenarioClickChart(scenarioList: ScenarioList): () => void,
+    onScenarioClickMap(scenario: Scenario): () => void
+}
+
+interface State {
+    children: Array<any>,
+    scenariosGraph: Array<any>
```
Comment (BenjiLee) [inline src/components/Filters/Scenarios.tsx:25]: Change to ScenarioList or Array<Scenario>

File: src/components/Filters/Scenarios.tsx [hunk 2]
```diff
@@ -1,10 +1,48 @@
-import React, { Component } from 'react';
-import { Select } from 'antd';
-import { formatTitle } from '../../utils/utils';
-import { styles } from '../../utils/constants';
+import React, {Component} from 'react';
+import {Select} from 'antd';
+import {formatTitle} from '../../utils/utils';
+import {styles} from '../../utils/constants';
+import {Scenario} from "../../utils/constantsTypes";
```
Comment (lxu213) [inline src/components/Filters/Scenarios.tsx:5]: I think your linter got over-excited again - if you don't have a preference, for continuity, let's keep the `{ Component }` format with spacing!

File: src/components/Filters/Scenarios.tsx [hunk 3]
```diff
@@ -15,21 +53,22 @@ class Scenarios extends Component {
     componentDidMount() {
         const children = [];
         const scenariosGraph = Array.from(this.props.SCENARIOS);
-        const { Option } = Select;
+        const {Option} = Select;
```
Comment (lxu213) [inline src/components/Filters/Scenarios.tsx:56]: same comment here

**Comment by BenjiLee** [comment]
@lxu213 Added linting in 066e874ccbf594bc1bb24a6bacaa8a60ffa3d88e

**Pull request merged**